resolving-dependency-issues.md

```diff
@@ -0,0 +1,254 @@
+
+Usually, these duplications are only visible in your package manager's lockfile and they either disappear or appear as you update your packages (meaning, their versions are bumped within their current version ranges), or when you upgrade packages (Note: This problem is somewhat more common with [Yarn v1 (Classic)](https://classic.yarnpkg.com/lang/en/) which doesn't deduplicate packages as actively as other package managers.)
+
+When running into transitive duplicates, this doesn't mean that your lockfile is broken or that you must delete it, and sometimes deleting it may not help due to your package manager's caches.
+Package managers have commands that can automatically deduplicate transitive dependencies, which is the most reliable way to align your dependencies:
+
+```sh
+# For npm:
+npm dedupe
+# For pnpm:
+pnpm dedupe
+# For Yarn Berry:
+yarn dedupe
+# For Yarn v1 (Classic)
+npx yarn-deduplicate && yarn
+```
+
+> [!NOTE]
+> Bun currently [does not yet have official support for deduplicating dependencies](https://github.com/oven-sh/bun/issues/8594#issuecomment-2195796840).
+> However, deleting your Bun lockfile will typically achieve deduplication reliably.
+
+You can observe the changes by `git diff`-ing your lockfile afterwards. You'll likely also see non-native modules being deduplicated. It's best practice to keep track of updating your dependencies regularly, as much as you would upgrade your top-level dependencies.
+
+```sh
+# For npm:
+npm update
+# For pnpm:
+pnpm update
+# For Yarn Berry:
+yarn up
+# For Yarn v1 (Classic)
+yarn upgrade
+```
+
+You can pass specific package names to these commands as well, if you need to selectively update packages.
+
+### 4. Overriding conflicting dependencies
+
+Lastly, some of your packages or dependencies may have conflicting `dependencies` or `peerDependencies` that you cannot resolve. For example, one package may specify an outdated peer dependency range for `react@^18.0.0`, while Expo requires `react@^19.0.0`. The same can happen with regular dependencies, and both JavaScript-only packages or native modules. This is even more common in monorepos that contain multiple apps.
+
+First, check if the affected packages can support and run with a newer (or older) version of the conflicting dependency. If they can, you can resolve this by forcing your package manager to install a specific version of the package. This is done by adding a resolution to your `package.json`:
+
+```json
+{
+  "name": "my-project",
+  "version": "0.0.0",
+  // ...
+  "resolutions": {
+    "react": "^19.1.0"
+  }
+}
+```
+
+For npm, you have to use a property named `overrides` rather than `resolutions`, but it's otherwise the same. Reinstall your dependencies after adding this resolution.
```

Comment:
I believe `npm` tends to be the default for beginners who are also more likely to be confused about dependency issues - for that reason, I think it would be worth writing out the npm example as well instead of adding this as a note.